onnxruntime/core/optimizer/insert_cast_transformer.cc

```diff
@@ -118,9 +118,9 @@ class RemoveDuplicateCastTransformer : public GraphTransformer {
         for (auto it = node.OutputNodesBegin(); it != node.OutputNodesEnd(); ++it) {
           const Node& output_node{*it};
           if (output_node.OpType() == "Cast") {
-            // Skip if the node's output is also the output of the graph
+            // Skip this child node if this child node's output is also an output of the graph
             if (graph_outputs.find(output_node.OutputDefs()[0]) != graph_outputs.end()) {
-              break;
+              continue;
```

Comment:
Need not break here - there could be other "child" nodes consuming the output of the current "Cast" node that could potentially be optimized away - breaking here would prevent this